### ISSUE TITLE
How do I set the Monitors on platforIo.NativePtr? 

### ISSUE THREAD

**Issue by ScottKane** [mellinoe/ImGui.NET#366]
You can't create a `new ImVector(numMonitors, numMonitors, data)` to assign to `platformIo.NativePtr->Monitors` as shown in the docking sample, also the following delegates are no longer available, how do I configure it now?

Platform_CreateWindow
Platform_DestroyWindow
Platform_GetWindowPos
Platform_ShowWindow
Platform_SetWindowPos
Platform_SetWindowSize
Platform_GetWindowSize
Platform_SetWindowFocus
Platform_GetWindowFocus
Platform_GetWindowMinimized
Platform_SetWindowTitle

Thanks

**Comment by mellinoe** [comment]
Take a look at the `docking` branch in this repository. It includes an up-to-date version of the sample project which renders with multi-viewports. Here's the relevant code which interacts with `ImGuiPlatformIO::Monitors`:

https://github.com/mellinoe/ImGui.NET/blob/docking/src/ImGui.NET.SampleProgram/ImGuiController.cs#L507-L534

**Comment by mellinoe** [comment]
Sorry for the quick response as you're already aware of the `docking` branch example code. However, the code I linked should still be working against the 1.88 native binary (the branch is up-to-date with `master`), and the functions in question are still being exposed by cimgui in the latest version as far as I can tell: https://github.com/cimgui/cimgui/blob/master/cimgui.cpp#L5004-L5026

**Comment by ScottKane** [comment]
The following code from the docking example is not available (here is no way to tell the platform how to get the window position/size): https://github.com/mellinoe/ImGui.NET/blob/docking/src/ImGui.NET.SampleProgram/ImGuiController.cs#L113

Also the code shown here is also no longer accurate: https://github.com/mellinoe/ImGui.NET/blob/docking/src/ImGui.NET.SampleProgram/ImGuiController.cs#L522

There is no constructor for ImVector that takes this information, not sure if this is a code gen issue or something but without an updated solution to setting the platform window position/size callbacks and assigning the Monitors ImVector I can't successfully replicate the old example.

I'm assuming the `Platform_` delegates aren't important and I can just create my own delegates with the same signature and everything should be fine but I can't test that without knowing how to address the 2 mentioned problems.

Thanks :)

**Comment by ScottKane** [comment]
Unless I'm missing something and I should be building the docking branch from source myself, I was just going with migrating from `ImGui.NET.Docking` version `1.75.0` to `ImGui.NET` version `1.88.0` (latest version available on nuget which I'm assuming is the master branch).

**Comment by ScottKane** [comment]
Ok I guess I was being stupid in thinking that docking features had been merged, I built the docking branch and managed to get things working. For anyone else needing access to the branch via nuget I have created `ImGui.NET.Branches.Docking`. Happy to transfer ownership of the package to you on nuget should you want/need it.

**Comment by mellinoe** [comment]
I haven't published any official NuGet packages called `ImGui.NET.Docking`, so you are using a package published by someone else, and it's unclear which version they have bundled into it. The latest published version of ImGui.NET has all of the docking support built into it, and the NuGet package itself doesn't have any of the sample / renderer code in it anyways. There's no difference in the ImGui bindings between the `master` and `docking` branch -- the only difference is in the sample code.

**Comment by ScottKane** [comment]
@mellinoe When I try to use the latest nuget build of `ImGui.NET`, it is missing the required platform code used by the `ImGuiController` which I have copied into my project

**Comment by ScottKane** [comment]
The latest version of ImGui.NET is missing the following methods from the docking branch:

```cs
ImGuiNative.ImGuiPlatformIO_Set_Platform_GetWindowPos(platformIo.NativePtr, Marshal.GetFunctionPointerForDelegate(_getWindowPos));
ImGuiNative.ImGuiPlatformIO_Set_Platform_GetWindowSize(platformIo.NativePtr, Marshal.GetFunctionPointerForDelegate(_getWindowSize));
```

And is missing the following delegates:

```
Platform_CreateWindow
Platform_DestroyWindow
Platform_GetWindowPos
Platform_ShowWindow
Platform_SetWindowPos
Platform_SetWindowSize
Platform_GetWindowSize
Platform_SetWindowFocus
Platform_GetWindowFocus
Platform_GetWindowMinimized
Platform_SetWindowTitle
```

But when I built the docking branch myself all of the above is present so there must be a difference between the branches

**Comment by ScottKane** [comment]
This can be worked around by creating the delegate types manually, importing the native platform functions and replacing the monitors code with:

```cs
var monitors = new ImVector<ImGuiPlatformMonitor>(numMonitors, numMonitors, data);
platformIo.NativePtr->Monitors = Unsafe.As<ImVector<ImGuiPlatformMonitor>, ImVector>(ref monitors);
```

**Comment by mellinoe** [comment]
> But when I built the docking branch myself all of the above is present so there must be a difference between the branches

You're right, there were certain native functions and delegates which were manually exposed in the docking branch but not in the master branch. I've corrected this now. It is not included in the 1.89.1 release, but you can build against master directly for the time being.
